scalableminds/webknossos
# Fix Wrapping Nested Tree Groups During Upload

Fixes a bug where nested tree groups were messed up during NML upload if “Create a new tree group for this file.“ is selected.

The group structure is a list of groups which can each have a list of children. When determining the largest free id, these children have to be taken into account as well.

### URL of deployed dev instance (used for testing):
- https://fixwrapnestedgroups.webknossos.xyz

### Steps to test:
- Upload an NML with nested groups, or one with un-connected trees within groups, like the one posted here https://scm.slack.com/archives/C5AKLAV0B/p1623331946023900?thread_ts=1623331839.023800&cid=C5AKLAV0B
- should upload fine, have stable group hierarchy
- should then download fine and re-upload fine.

### Issues:
- fixes #5495

------
